tx_metric/tx_metric.go

```diff
@@ -51,7 +59,12 @@ func (tm *TxMetric) Start() error {
 
 		if bl.Time() <= tm.StopTime {
```

Comment:
it is not at the PR, but we can add an early break with condition bl.Time() > tm.StopTime

Reply:
We had 
`} else {
			break
		}`
so I think adding an early break is not necessary.